ConsumerAffairs/django-locking
# Compatibility with Django 1.9 - 1.11

Dropped support for older versions.

Added sample project for easy demo.

## Practical testing instructions

These are practical instructions for developers.

### Setup

1. Checkout the repo
1. Create a virtualenv for it with Python 2.7.x
1. `cd` into the repo root directory
1. install the package by running `pip install -e .`
1. Run `cd sample_project`
1. Run the migrations: `./manage.py migrate`
1. Create two superusers by running `./manage.py createsuperuser` twice

### Tests
- [x] Test 1. - Check that locking the admin pages is working 
1. Log in to the admin
1. Create two new `Poll` entries.
1. Enter the "Change" page in the admin for one of the `Poll` entries
1. In another browser, or another browser mode (private, guest, incognito...), log in to the admin with the other account.
1. Open the Polls change list and verify that there is a padlock icon next to the entry being edited in the other browser.
1. Open this entry and verify that the page is read-only with a big red message at the top explaining that another user is currently editing this entry.

- [x] Test 2 - Run unit tests
1. In the shell, run unit tests by `./manage.py test locking`.
1. Verify that there are no errors or failures.

- [x] Test 3 - Run unit tests with Django 1.9
1. In the shell, install the earlier version of Django by running `pip install "django>1.8,<1.10"`
1. Run unit tests by `./manage.py test locking`.
1. Verify that there are no errors or failures.
